doc/source/user_guide/sparse.rst

```diff
@@ -105,84 +130,117 @@ keeps an arrays of all of the locations where the data are not equal to the
-* ``bool``: ``False``
+
+The *ufunc* is also applied to ``fill_value``. This is needed to get
+the correct dense result.
 
 .. ipython:: python
 
-   s = pd.Series([1, np.nan, np.nan])
-   s
-   s.to_sparse()
+   arr = pd.SparseArray([1., -1, -1, -2., -1], fill_value=-1)
+   np.abs(arr)
+   np.abs(arr).to_dense()
 
-   s = pd.Series([1, 0, 0])
-   s
-   s.to_sparse()
+.. _sparse.migration:
 
-   s = pd.Series([True, False, True])
-   s
-   s.to_sparse()
+Migrating
+---------
+
+In older versions of pandas, the ``SparseSeries`` and ``SparseDataFrame`` classes (documented below)
+were the preferred way to work with sparse data. With the advent of extension arrays, these subclasses
+are no longer needed. Their purpose is better served by using a regular Series or DataFrame with
+sparse values instead.
+
+**There's no performance or memory penalty to using a Series or DataFrame with sparse values,
+rather than a SparseSeries or SparseDataFrame**.
+
+This section provides some guidance on migrating your code to the new style. As a reminder, you can
+use the python warnings module to control warnings. If you wish to ignore the warnings,
+
+.. code-block:: python
+
+   >>> import warnings
+
+   >>> warnings.filterwarnings('ignore', 'Sparse', FutureWarning)
+   >>> pd.SparseSeries()  # No warning message
+   Series([], dtype: Sparse[float64, nan])
+   BlockIndex
+   Block locations: array([], dtype=int32)
+   Block lengths: array([], dtype=int32)
+
+But we recommend modifying your code, rather than ignoring the warning.
+
+**Construction**
 
-You can change the dtype using ``.astype()``, the result is also sparse. Note that
-``.astype()`` also affects to the ``fill_value`` to keep its dense representation.
+From an array-like, use the regular :class:`Series` or
+:class:`DataFrame` constructors with :class:`SparseArray` values.
 
+.. code-block:: python
+
+   # Old way
+   >>> pd.SparseDataFrame({"A": [0, 1]})
 
 .. ipython:: python
 
-   s = pd.Series([1, 0, 0, 0, 0])
-   s
-   ss = s.to_sparse()
-   ss
-   ss.astype(np.float64)
+   # New way
+   pd.DataFrame({"A": pd.SparseArray([0, 1])})
 
-It raises if any value cannot be coerced to specified dtype.
+From a SciPy sparse matrix, use :meth:`DataFrame.sparse.from_spmatrix`,
 
-.. code-block:: ipython
+.. code-block:: python
 
-   In [1]: ss = pd.Series([1, np.nan, np.nan]).to_sparse()
-   Out[1]:
-   0    1.0
-   1    NaN
-   2    NaN
-   dtype: float64
-   BlockIndex
-   Block locations: array([0], dtype=int32)
-   Block lengths: array([1], dtype=int32)
+   # Old way
+   >>> from scipy import sparse
+   >>> mat = sparse.eye(3)
+   >>> df = pd.SparseDataFrame(mat, columns=['A', 'B', 'C'])
 
-   In [2]: ss.astype(np.int64)
-   Out[2]:
-   ValueError: unable to coerce current fill_value nan to int64 dtype
+.. ipython:: python
 
-.. _sparse.calculation:
+   # New way
+   from scipy import sparse
+   mat = sparse.eye(3)
+   df = pd.DataFrame.sparse.from_spmatrix(mat, columns=['A', 'B', 'C'])
+   df
```

Comment:
maybe add `df.dtypes` ? (since the df repr does not give any indication of its sparseness)